File: back-end/demo/src/main/java/com/album/react/board/model/dto/Board.java
package com.album.react.board.model.dto;

import com.album.react.user.model.dto.User;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Board {
	
	private int boardNo;
	private String boardTitle;
	private String boardContent; 
	private String writeDate; 
	private int likeCount; 
	private int viewCount; 
	private int commentCount;
	
	private int userNo; 
	private String writerName; 
	private String writerProfileImg; 
	private String userEmail; 

}

File: front-end/src/pages/index/types/board.ts
export default interface Board {
  boardNo: number;
  boardTitle: string;
  boardContent: string;
  likeCount: number;
  commentCount: number;
  viewCount: number;
  writeDate: string;
  writerName: string;
  writerProfileImg: string | null;
  boardImg: string | null;
  userEmail: string;
}

export interface Comment {
  userName: string;
  userProfileImg: string;
  commentContent: string;
  commentNo: number;
  boardNo: number;
  parentCommentNo?: number | null;
}
